node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -74,40 +74,41 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
     * object. Internally in [[org.bitcoins.node.networking.P2PClient p2p client]] you will see that
     * the [[ChainApi chain api]] is updated inside of the p2p client
     */
-  lazy val client: P2PClient = {
-    val peerMsgRecv: PeerMessageReceiver =
-      PeerMessageReceiver.newReceiver(node = this, peer = peer)
-    val p2p = P2PClient(context = system,
+  lazy val clients: Vector[P2PClient] = {
+    val peerMsgRecvs: Vector[PeerMessageReceiver] = peers.map(x=>
+      PeerMessageReceiver.newReceiver(node = this, peer = x))
+    val p2p =  peers zip peerMsgRecvs map {case(peer,peerMsgRecv)=> P2PClient(context = system,
```

Comment:
Let's avoid using the "fancy" scala features where you can omit `.` notation. It may be easier for non engineers to read, but for engineers on larger code bases it can get complicated. I think they might be requiring dot notation (`.`) in scala3 too. 

Here is the re-write 

```scala
    val peerMsgRecvs: Vector[PeerMessageReceiver] = peers.map { x =>
      PeerMessageReceiver.newReceiver(node = this, peer = x)
    }
    val zipped = peers.zip(peerMsgRecvs)
    val p2p =  zipped.map { case(peer,peerMsgRecv)=> 
      P2PClient(context = system,
                        peer = peer,
                        peerMessageReceiver = peerMsgRecv)
    }
    p2p
```